stores/json/store.go

```diff
@@ -72,14 +73,22 @@ func (store Store) sliceFromJSONDecoder(dec *json.Decoder) ([]interface{}, error
 
 var errEndOfObject = fmt.Errorf("End of object")
 
+func correctKeyForEncoding(key string) string {
+	// handles escaping in the original json
+	if strings.Contains(key, "\\") {
+		key = strings.Replace(key, "\\", "\\\\", -1)
```

Comment:
two questions:
1. what does the json spec say about backslash encoding?
2. shouldn't the end-users take care of this before calling sops instead of doing magic here?

Reply:
There are plenty of other cases other than "\\" that need escaping as per [RFC 7159](https://tools.ietf.org/html/rfc7159#section-7).

I think the correct way to solve this would be to fix this https://github.com/ajvb/sops/blob/58524f7d7990d7cc8c2509d79446d912d5e590cf/stores/json/store.go#L180 so that it JSON-encodes strings before outputting.

Reply:
Thanks for asking these questions. In reading the spec, I tested out how `sops -e` functions with other escaping scenarios (like `"a_key_with\"quotes\"": "value"`) and it broke in the same way. I think what I just pushed is closer to what we want.

1. In looking at [RFC 7159#Strings](https://tools.ietf.org/html/rfc7159#section-7) and [ECMA-404](https://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf), they just call out that there are a few characters (like `"` and `\`) that need to be escaped
2. I don't think so, imo we should decode and encode compliant json. The examples in #356 are strange to me, but they are proper json keys.

Reply:
I wonder if we should encode values like this too. I think we are emitting invalid JSON when we output decrypted JSON files if we do it the same way we did for the keys, as the encrypted values could contain characters that need to be escaped too, right?

Reply:
@autrilla ah, missed your first comment. Yeah totally agree.

In regards to values, we are already encoding using `json.Marshal` - https://github.com/mozilla/sops/blob/master/stores/json/store.go#L138 - that's actually where I realized that needed to be applied to keys as well.

Reply:
Oh, hah, okay, we already do the right thing for values!